update_prevalence.py

```diff
@@ -281,16 +281,53 @@ def recent_daily_cumulative_cases(self) -> List[int]:
     # cases. Nominally is values[-1] - values[0], but sometimes regions post
     # corrections which result in the number of cases decreasing.
     def cases_in_cum_cases(self, values) -> int:
-        min_index = values.index(min(values))
-        return max(values[min_index:]) - min(values)
+        # list of indices right before negative corrections. If values = [3,2,3,0,5],
+        # then negative_corrections == [0, 2]
+        negative_corrections = [i for i, val in enumerate(values[:-1]) if val > values[i+1]]
+        corrections_within_bounds = min(values) == values[0] and max(values) == values[-1]
+        if len(negative_corrections) == 0 or corrections_within_bounds:
+            return values[-1] - values[0]
+
+        # Always use values[-1] rather than max(values) because max(values) is
+        # either values[-1] or there's been a negative correction that should take precedence.
+        if min(values[:-1]) <= values[-1]:
+            possibly_suspect_correction = False
+            for local_max_idx in negative_corrections:
+                value_before_correction = values[local_max_idx]
+                value_correction = values[local_max_idx + 1]
+                value_after_correction = None if local_max_idx + 2 >= len(values) else values[local_max_idx + 2]
+
+                if value_after_correction is not None:
+                    correction_size = value_before_correction - value_correction # flipped
+                    overall_change = value_after_correction - value_before_correction
```

Comment:
Is this basically saying "was the correction reversed out the next day", right? I'd expect "overall change" to mean the difference from day -7 to day 0.